src/UI/TransactionsToastList/index.tsx

```diff
@@ -103,6 +119,39 @@ export const TransactionsToastList = ({
     )
   );
 
+  const clearNotPendingTransactionsFromStorage = (e: BeforeUnloadEvent) => {
+    e.preventDefault();
+
+    const toasts = transactionToastsSelector(store.getState());
```

Comment:
I think we should move this logic to the TransactionToast component, since we have all the data there and we don't need to re-read everything on umount

Reply:
I do not agree with this. I want to keep a Single Responsibility for the Toast component.